extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleReplaceShowObjectCommands.scala

```diff
@@ -101,3 +103,24 @@ case class FilteredShowFunctionsCommand(delegated: RunnableCommand)
     result != null && result.getIsAllowed
   }
 }
+
+case class FilteredShowColumnsCommand(delegated: RunnableCommand)
+  extends FilteredShowObjectCommand(delegated) with WithInternalChild {
+
+  override val output: Seq[Attribute] = delegated.output
+
+  override def run(spark: SparkSession): Seq[Row] = {
+    val rows = delegated.run(spark)
+    val table = delegated.asInstanceOf[ShowColumnsCommand].tableName
+    val ugi = AuthZUtils.getAuthzUgi(spark.sparkContext)
+    rows.filter(f =>
+      isAllowed(Row(table.database.getOrElse("default"), table.table, f.getString(0)), ugi))
```

Comment:
this is a behavior change, why forced it to default? the database is optional for temp views

Reply:
yes